### PR TITLE
Update SQL Bindings API

### PR DESCRIPTION
**Pull request by barbaravaldez**
Refactor the AddSqlBindingQuickpick function and create different methods to reuse later.

### CODE REVIEW

File: extensions/sql-bindings/src/sql-bindings.d.ts [hunk 1]
```diff
@@ -66,7 +86,7 @@ declare module 'sql-bindings' {
 	/**
 	 * Azure Functions binding type
 	 */
-	export const enum BindingType {
+	export enum BindingType {
```
Comment (Charles-Gagnon) [inline extensions/sql-bindings/src/sql-bindings.d.ts:89]: You can't have non-const enums in typings files, why did you remove this?

**Review by Charles-Gagnon — CHANGES_REQUESTED** [review]
When making changes like this it's good to include why you need to make these changes. It's not clear to me why this "SqlBindingConfig" class was added

File: extensions/sql-bindings/src/sql-bindings.d.ts (hunk 1 above)
Comment (barbaravaldez) [reply extensions/sql-bindings/src/sql-bindings.d.ts:89]: I did this to only return the BindingType on the promptSqlBindingMethod, instead of returning the quickPickItem as well. We use the label to send a telemetry event.

File: extensions/sql-bindings/src/dialogs/addSqlBindingQuickpick.ts [hunk 2]
```diff
@@ -5,294 +5,310 @@
...
+import { ConnectionDetails, IConnectionInfo, IExtension } from 'vscode-mssql';
 import * as constants from '../common/constants';
 import * as utils from '../common/utils';
 import * as azureFunctionsUtils from '../common/azureFunctionsUtils';
 import { TelemetryActions, TelemetryReporter, TelemetryViews } from '../common/telemetry';
 import { BindingType } from 'sql-bindings';
 
-export async function launchAddSqlBindingQuickpick(uri: vscode.Uri | undefined): Promise<void> {
-	TelemetryReporter.sendActionEvent(TelemetryViews.SqlBindingsQuickPick, TelemetryActions.startAddSqlBinding);
+export class SqlBindingConfig {
```
[the remark this answers is not in the document]
Comment (barbaravaldez) [reply extensions/sql-bindings/src/dialogs/addSqlBindingQuickpick.ts:15]: I created a class, because I thought we would use the mssql api in different functions, but it seems that it's only used when getting the connection string setting name

File: extensions/sql-bindings/src/dialogs/addSqlBindingQuickpick.ts [hunk 3]
```diff
@@ -5,294 +5,310 @@
...
 
-		let existingSettings: (vscode.QuickPickItem)[] = [];
-		if (settings?.Values) {
-			existingSettings = Object.keys(settings.Values).map(setting => {
-				return {
-					label: setting
-				} as vscode.QuickPickItem;
-			});
-		}
+	public async promptForBindingType(): Promise<BindingType | undefined> {
```
Comment (VasuBhog) [inline extensions/sql-bindings/src/dialogs/addSqlBindingQuickpick.ts:112]: I would say you can move these prompts to azureFunctionUtils - so that we could leverage them in other places as well.

File: extensions/sql-bindings/src/common/azureFunctionsUtils.ts [hunk 4]
```diff
@@ -307,3 +309,207 @@ export async function getAFProjectContainingFile(fileUri: vscode.Uri): Promise<v
...
+export async function promptForObjectName(bindingType: BindingType): Promise<string | undefined> {
+	return vscode.window.showInputBox({
+		prompt: bindingType === BindingType.input ? constants.sqlTableOrViewToQuery : constants.sqlTableToUpsert,
+		placeHolder: constants.placeHolderObject,
+		validateInput: input => input ? undefined : constants.nameMustNotBeEmpty,
+		ignoreFocusOut: true
+	});
+}
+
+export async function promptForConnectionStringSetting(projectUri: vscode.Uri | undefined): Promise<string | undefined> {
```
Comment (Charles-Gagnon) [inline extensions/sql-bindings/src/common/azureFunctionsUtils.ts:343]: `promptAndUpdateConnectionStringSetting` - this isn't just prompting so we should be clear here

**Review by Charles-Gagnon — APPROVED** [review]

**Review by VasuBhog — APPROVED** [review]
